### PR TITLE
test: Use HTTP retries on external connections if needed

### PR DESCRIPTION
**Pull request by jrajahalme**
Recently introduced tests using external sites cause intermittent CI failures. Enable Envoy HTTP retries for those tests.

Note that the current Envoy retries do not work properly on upstream connections which re-use the original source address and port, so this change alone would not fix these intermittent CI failures, but will not make them any worse either. Envoy side fix of retries on these cases is prepared separately.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

```release-note
helm: Add helm template option to set the upstream HTTP retry timeout for L7 HTTP visibility and enforcement (e.g., `--set global.proxy.httpRetryTimeout=7`). Without this option retries are only enabled for immediate connection errors. With this option retries are started after the upstream server does not respond within the given number of seconds.
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9892)

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Review by soumynathan — COMMENTED** [review]
Looks good to me and based on the review comments, it seems that we need a release note.

File: install/kubernetes/cilium/values.yaml
Comment (aanm) [inline install/kubernetes/cilium/values.yaml:75]: Please add a small description here for what this field is used for

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
running tests with all k8s versions to see if the flake is hit

**Comment by jrajahalme** [comment]
@aanm `swapi.co` timeout was hit in k8s 1.15 tests. Note that while this PR enables retries, due to an upstream Envoy "unforeseen feature interaction" between retries and use of the original source address (and port) in upstream connections, retries do not work right now. IMO we can either merge this PR now, or wait for the Envoy fix to be ready and merge this after cilium-envoy is updated.

File: install/kubernetes/cilium/values.yaml
Comment (jrajahalme) [reply install/kubernetes/cilium/values.yaml:75]: Added, also added support for the rest of the cilium-agent HTTP options, not setting them by default, but testing setting all of them in the CI test.

**Comment by jrajahalme** [comment]
test-me-please

**Review by aanm — CHANGES_REQUESTED** [review]
@jrajahalme it seems this PR triggers some helm warnings:

```
coalesce.go:199: warning: destination for proxy is a table. Ignoring non-table value <nil>
coalesce.go:133: key proxy is table. Skipping
coalesce.go:133: key proxy is table. Skipping
coalesce.go:133: key proxy is table. Skipping
coalesce.go:133: key proxy is table. Skipping
coalesce.go:133: key proxy is table. Skipping
coalesce.go:133: key proxy is table. Skipping
coalesce.go:199: warning: destination for proxy is a table. Ignoring non-table value <nil>
```

probably because the fields in the helm charts are commented?

**Comment by tgraf** [comment]
Waiting on upstream Envoy to enable the required functionality

**Pull request closed without merging**
